python/plugins/processing/algs/qgis/Dissolve.py

```diff
@@ -49,17 +50,24 @@ class Dissolve(GeoAlgorithm):
 
     def processAlgorithm(self, progress):
         useField = not self.getParameterValue(Dissolve.DISSOLVE_ALL)
-        fieldname = self.getParameterValue(Dissolve.FIELD)
         vlayerA = dataobjects.getObjectFromUri(
             self.getParameterValue(Dissolve.INPUT))
-        field = vlayerA.fieldNameIndex(fieldname)
         vproviderA = vlayerA.dataProvider()
-        fields = vproviderA.fields()
+        fields = QgsFields()
+        if useField:
+            fieldname = self.getParameterValue(Dissolve.FIELD)
+            fieldIdx = vlayerA.fieldNameIndex(fieldname)
+            field = vlayerA.fields()[fieldIdx]
```

Comment:
Am 31.08.2015 um 09:11 schrieb Matthias Kuhn:
> In python/plugins/processing/algs/qgis/Dissolve.py
> <https://github.com/qgis/QGIS/pull/2263#discussion_r38288863>:
>
>>          vproviderA = vlayerA.dataProvider()
>> -        fields = vproviderA.fields()
>> +        fields = QgsFields()
>> +        if useField:
>> +            fieldname = self.getParameterValue(Dissolve.FIELD)
>> +            fieldIdx = vlayerA.fieldNameIndex(fieldname)
>> +            field = vlayerA.fields()[fieldIdx]
>
> Stability is increased by not mixing /provider/ and /layer/ indexes
> which may differ.

Agreed, as I wrote before: I have no idea why I used provider in the 
first place (I always use QgsVectorLayer.fields()[fieldIndex]

>
> And I would say it increases readability

recoded

>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qgis/QGIS/pull/2263/files#r38288863>.
>



__________ Information from ESET Mail Security, version of virus signature database 12176 (20150831) __________

The message was checked by ESET Mail Security.
http://www.eset.com